src/PrestaShopBundle/Twig/AdminExtension.php

```diff
@@ -94,7 +96,21 @@ public function initRuntime(\Twig_Environment $environment)
                 // Get current route name to know when to put "current" class on HTML dom
                 $currentRouteName = $parameterBag->get('_route');
 
+                $context = (new LegacyContext())->getContext();
+                $translator = $context->getTranslator();
+                $locale = $context->language->locale;
```

Comment:
No. If you need the translator, use the `$container` already sent by the dependency injection.

Then, you can get your locale from it and do not require the `LegacyContext`. You will see that it also fixes your tests.
```php
$locale = $this->container->get('translator')->getLocale();
```

But this will return something like `fr-FR`, which may not fit your needs. In that case, you can use this code to separate the country code from the language code.
```php
list($isoLang, $countryCode) = explode('-', $this->container->get('translator')->getLocale());
```

This is already done in several classes related to the modules, as you can see in: https://github.com/PrestaShop/PrestaShop/blob/76288006b6944101f4fb4b20744a172bf5b977d5/src/Core/Addon/Module/ModuleRepository.php#L119